packages/blade/src/components/ButtonGroup/__tests__/ButtonGroup.web.test.tsx

```diff
@@ -0,0 +1,196 @@
+import { ChevronDownIcon, PlusIcon } from '~components/Icons';
+import { Dropdown, DropdownButton, DropdownOverlay } from '~components/Dropdown';
+import { ActionList, ActionListItem } from '~components/ActionList';
+import { AutoComplete } from '~components/Input/DropdownInputTriggers';
+
+beforeAll(() => jest.spyOn(console, 'error').mockImplementation());
+afterAll(() => jest.restoreAllMocks());
+
+const variants: ButtonGroupProps['variant'][] = ['primary', 'secondary', 'tertiary'];
+const colors: ButtonGroupProps['color'][] = ['primary', 'white', 'positive', 'negative'];
+const sizes: ButtonGroupProps['size'][] = ['xsmall', 'small', 'medium', 'large'];
+
+describe('<ButtonGroup />', () => {
+  it('should render ButtonGroup with default properties', () => {
+    const { container, getByRole, getAllByRole } = renderWithTheme(
+      <ButtonGroup>
+        <Button>One</Button>
+        <Button>Two</Button>
+        <Button>Three</Button>
+      </ButtonGroup>,
+    );
+    expect(container).toMatchSnapshot();
+    expect(getByRole('group')).toBeInTheDocument();
+
+    const [button1, button2, button3] = getAllByRole('button');
+    expect(button1).toHaveTextContent('One');
+    expect(button2).toHaveTextContent('Two');
+    expect(button3).toHaveTextContent('Three');
+  });
+
+  it('should render ButtonGroup with dropdown', () => {
+    const { container, getByRole, getAllByRole } = renderWithTheme(
+      <ButtonGroup>
+        <Button icon={PlusIcon}>Payout</Button>
+        <Dropdown>
+          <DropdownButton icon={ChevronDownIcon}> More</DropdownButton>
+          <DropdownOverlay>
+            <ActionList>
+              <ActionListItem title="Bulk Payout" value="bulk-payout" />
+              <ActionListItem title="Upload Invoice" value="upload-invoice" />
+              <ActionListItem title="Add Contact" value="add-contact" />
+              <ActionListItem title="Team Member" value="team-member" />
+            </ActionList>
+          </DropdownOverlay>
+        </Dropdown>
+      </ButtonGroup>,
+    );
+    expect(container).toMatchSnapshot();
+    expect(getByRole('group')).toBeInTheDocument();
+
+    const [button1, button2] = getAllByRole('button');
+    expect(button1).toHaveTextContent('Payout');
+    expect(button2).toHaveTextContent('More');
+  });
+
+  sizes.forEach((size) => {
+    it(`should render ButtonGroup with ${size} size`, () => {
+      const { container } = renderWithTheme(
+        <ButtonGroup size={size}>
+          <Button>One</Button>
+          <Button>Two</Button>
+          <Button>Three</Button>
+        </ButtonGroup>,
+      );
+      expect(container).toMatchSnapshot();
+    });
+  });
+
+  it('should render ButtonGroup with full width', () => {
+    const { container } = renderWithTheme(
+      <ButtonGroup isFullWidth={true}>
+        <Button>One</Button>
+        <Button>Two</Button>
+        <Button>Three</Button>
+      </ButtonGroup>,
+    );
+    expect(container).toMatchSnapshot();
```

Comment:
maybe use `.toHaveStyle` to assert over specific style instead of full snapshot? lots of snapshots in this test

Reply:
I feel snapshots are fine, Button tests also have a similar pattern.